mbaldessari/connected-vehicle-architecture
# common automatic update

- Add support for /values-<platform>-<clustergroup>.yaml
- Use Kafka version 3.1.0
- Use newer car-simulator tag
- Fix up tests after new changes
- Fix gitleaks file after last unit test change
- Fix up .gitleaks.toml file
- Allow multiple test inputs
- Initial v2 plumbing
- Add module_utils path for shared common code
- Move v1 secret functions in module_utils
- Split out common function in separate file
- Move the secret format implementation in classes
- A bunch of cleanups of unit-testing. Still not fully passing
- Have super-linter pass
- Fix patching the right function
- Start unit testing V2 as well
- Add version check in v1 implementation
- Force get_version() to return a string for the time being
- Pass proper parameters to V2 class
- Initial start of vault policies implementation
- More validation and more V2 testing
- Make sure we check if the vaultPolicy exists
- Make error message clearer
- Add a lot more tests and start implementing the files: section
- Fix black errors
- Cleanups and also call check_missing_secrets only on v1.0
- Fix black errors
- Add a test to cover for empty vaultPrefixes
- Add support and tests for the password generation via policy
- Actually use the proper oc exec call to generate secrets
- Implement file uploading as base64 attribute as well
- Move more logic into _get_file_path
- Large rewrite to unify fields and files and add initial base64 support and tests
- Add base64 support for secrets as well and fix up tests
- POC mostly working at this point
- For the time being always stay on getpass.getpass() for input
- Remove unused modules
- Add checks for duplicate secret names and field names
- Switch to using AnsibleModule.run_command
- Make the secret prompting a little prettier
- Implement looking for ~/values-secret-<patternname>.yaml first
- Test base64 secret
- Silence yaml indentation linting errors and more
- Added values-secret examples
- Drop callbacks_enabled
- Add another relevant change in Changes.md
- Error out nicely if yaml.safe_load() cannot parse the file
- Add support for ansible-vault
- Clean up comments in test-cases files and implement backingStore
- Use sensible prompt strings
- Be clearer in the text about ansible-vault
- Move checks that are specific to the version inside classes
- Add a test for erroring out when values_secret_template is not defined and check_missing_secrets is set to true
- Add a default validatedPatternDefaultPolicy password policy
- Forgot to add new test for default vp-policy
- Make sure we re-attempt all vault commands in case of failure
- Add some more documentation
- Add initial v2 schema json file
- Rename the description field to prompt as that is what it is really used for
- value and path can be 'null' when onMissingValue is set to 'prompt'
- Make version a string all the time
- Require some more fields
- Add some more examples to the schema file
- Test json schema CI job
- Fix small ansible lint warning
- Also try {{ pattern_dir }}/values-secret.yaml.template if all others failed
- Fix Changes.md with a small nit
- Prompt ansible-vault password via ansible.builtin.pause
- Make the tasks text less intimidating
- Add json schema for the v1 secret format
- Allow picking a custom values-secret.yaml file via the VALUES_SECRET env var
- Add filter_plugins path in common
-  Drop ansible/roles/bootstrap
- Add filter_plugin path
- Add override attribute support
- Move to draft-07 of the json schema for V2
- Do not raise exception if secret's existence check fails
- Upgrade vault-helm to v0.23.0
- Filter out patches to test/
- Add patch to be able to use vault-ssl as a default
- Switch to vault-ssl as a default
- Remove stable default from channels
- Split secret counter in two variables
- Fix common tests
- Move to secrets v2.0
- Remove unused values-region-one.yaml file
- Change channel to 7.10.x in amq-broker-rhel8 as 7.x seems to not exist anymore
- Move testing targets after the installation ones
- Move some of the oc calls into the legacy target
- Simplify getting the remote git URL associated to the target origin
- Drop legacy-install from common
- Move comments in the right place
- Clean up PHONY targets in Makefile
- Move all testing-related variables closer to the tests target
- Make sure we error out after the legacy removal
- Comment out namespace entry in values-hub.yaml file, it has no effect
- Drop legacy-install entirely
- drop insecureUnsealVaultInsideCluster entirely
- Fix up common/ tests
- Clean up Makefile
